pkg/archive/archive_linux_test.go

```diff
@@ -93,9 +93,11 @@ func checkFileMode(t *testing.T, path string, perm os.FileMode) {
 }
 
 func TestOverlayTarUntar(t *testing.T) {
-	oldmask, err := system.Umask(0)
+	oldMask, err := system.Umask(0)
 	require.NoError(t, err)
-	defer system.Umask(oldmask)
+	defer func(oldMask int) {
+		_, _ = system.Umask(oldMask) // Ignore err. This can only fail with ErrNotSupportedPlatform, in which case we would have failed above.
+	}(oldMask)
```

Comment:
(This can reference the outer variable directly, without the explicit copy in a parameter. The version in the PR works fine.

If you change this, please do it in the other 2 copies as well.)

Reply:
By “reference directly” I meant
```go
oldMask = …
defer func() {
   …(oldMask)
}()
```

The compiler automatically does the (logical equivalent of) passing `&oldMask` into the closure.